addons/velero/1.6.2/install.sh

```diff
@@ -162,13 +227,142 @@ function velero_patch_http_proxy() {
     [ "$(kubectl get sc longhorn -o jsonpath='{.metadata.annotations.storageclass\.kubernetes\.io/is-default-class}')" = "true" ]; then
-        render_yaml_file "$src/tmpl-change-storageclass.yaml" > "$dst/change-storageclass.yaml"
-        if [ -z "$disable_kustomization" ]; then
-            insert_resources "$dst/kustomization.yaml" change-storageclass.yaml
+
+        # when re-applying the same velero version, this might not exist.
+        if [ ! -f "$dst/kustomization.yaml" ]; then
+            cat > "$dst/kustomization.yaml" <<EOF
+namespace: ${VELERO_NAMESPACE}
+
+resources:
+EOF
         fi
+
+        render_yaml_file "$src/tmpl-change-storageclass.yaml" > "$dst/change-storageclass.yaml"
+        insert_resources "$dst/kustomization.yaml" change-storageclass.yaml
+
+    fi
+}
+
+function velero_migrate_from_object_store() {
+    local src="$1"
+    local dst="$2"
+
+    # TODO (dans): remove this feature flag when/if we decide to ship migration
+    if [ -z "$BETA_VELERO_MIGRATE_FROM_OBJECT_STORE" ]; then 
+        return
+    fi
+
+    echo ****HERE*****
+    # if there is still an object store, don't migrate. If KOTSADM_DISABLE_S# is set, force the migration
+    if [ ! "$KOTSADM_DISABLE_S3" == 1 ] || [ -n "$ROOK_VERSION" ] || [ -n "$MINIO_VERSION" ]; then 
+        echo ****EXIT*****
+        return
+    fi
+    echo ****CONTINUE*****
+
+    # if an object store isn't installed don't migrate
+    # TODO (dans): this doeesn't support minio in a non-standard namespace
+    if (! kubernetes_resource_exists rook-ceph deployment rook-ceph-rgw-rook-ceph-store-a) && (! kubernetes_resource_exists minio deployment minio); then 
+        return
+    fi
+
+    echo ****WOW*****
+
+    export VELERO_S3_HOST=
+    export VELERO_S3_ACCESS_KEY_ID=
+    export VELERO_S3_ACCESS_KEY_SECRET=
+    if kubernetes_resource_exists rook-ceph deployment rook-ceph-rgw-rook-ceph-store-a; then 
+        echo "Previous installation of Rook Ceph detected."
+        VELERO_S3_HOST="rook-ceph-rgw-rook-ceph-store.rook-ceph"
+        VELERO_S3_ACCESS_KEY_ID=$(kubectl -n rook-ceph get secret rook-ceph-object-user-rook-ceph-store-kurl -o yaml | grep AccessKey | head -1 | awk '{print $2}' | base64 --decode)
+        VELERO_S3_ACCESS_KEY_SECRET=$(kubectl -n rook-ceph get secret rook-ceph-object-user-rook-ceph-store-kurl -o yaml | grep SecretKey | head -1 | awk '{print $2}' | base64 --decode)
+    else 
+        echo "Previous installation of Minio detected."
+        VELERO_S3_HOST="minio.minio"
+        VELERO_S3_ACCESS_KEY_ID=$(kubectl -n minio get secret minio-credentials -ojsonpath='{ .data.MINIO_ACCESS_KEY }' | base64 --decode)
+        VELERO_S3_ACCESS_KEY_SECRET=$(kubectl -n minio get secret minio-credentials -ojsonpath='{ .data.MINIO_SECRET_KEY }' | base64 --decode)
+    fi
+
+    # If this is run through `velero_already_applied`, we need to create base kustomization file
+    if [ ! -f "$dst/kustomization.yaml" ];then
```

Comment:
how about moving this to velero_already_applied?

Reply:
I think the problem I'm running into is that in velero_already_applied I don't really even know if I'll need this or if it already exists. There are three states - no kustmoization necessary, just the configmap to restore from rook to longhorn, or migration + configmap. Open to refactoring but seems like I would have to create a lot more code to check all the conditions first before creating a kustomize file.